homeassistant/components/eq3btsmart/climate.py

```diff
@@ -17,141 +28,441 @@
     SUPPORT_TARGET_TEMPERATURE,
 )
 from homeassistant.const import (
+    ATTR_FRIENDLY_NAME,
     ATTR_TEMPERATURE,
     CONF_DEVICES,
     CONF_MAC,
+    CONF_UNIQUE_ID,
+    EVENT_HOMEASSISTANT_START,
     PRECISION_HALVES,
+    PRECISION_TENTHS,
+    PRECISION_WHOLE,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
     TEMP_CELSIUS,
 )
+from homeassistant.core import callback
 import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import async_generate_entity_id
+from homeassistant.helpers.event import async_track_state_change
+from homeassistant.helpers.reload import async_setup_reload_service
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.util.dt import utcnow
+
+from . import DOMAIN, PLATFORMS
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_SENSOR = "target_sensor"
+CONF_SENSOR_CONTROL = "control_by_target_sensor"
+CONF_MIN_TEMP = "min_temp"
+CONF_MAX_TEMP = "max_temp"
+CONF_PRECISION = "precision"
+CONF_COLD_TOLERANCE = "cold_tolerance"
+CONF_HOT_TOLERANCE = "hot_tolerance"
+CONF_OFF_TEMP_OFFSET = "off_temp_offset"
+CONF_ON_TEMP_OFFSET = "on_temp_offset"
+
+DEFAULT_TOLERANCE = 0.3
 STATE_BOOST = "boost"
 
 ATTR_STATE_WINDOW_OPEN = "window_open"
 ATTR_STATE_VALVE = "valve"
 ATTR_STATE_LOCKED = "is_locked"
 ATTR_STATE_LOW_BAT = "low_battery"
 ATTR_STATE_AWAY_END = "away_end"
+ATTR_STATE_THERMOSTAT = "thermostat_state"
+ATTR_STATE_ERROR = "error"
+ATTR_STATE_LAST_ERROR_AT = "last_error_at"
 
 EQ_TO_HA_HVAC = {
-    eq3.Mode.Open: HVAC_MODE_HEAT,
     eq3.Mode.Closed: HVAC_MODE_OFF,
+    eq3.Mode.Open: HVAC_MODE_HEAT,
     eq3.Mode.Auto: HVAC_MODE_AUTO,
     eq3.Mode.Manual: HVAC_MODE_HEAT,
-    eq3.Mode.Boost: HVAC_MODE_AUTO,
     eq3.Mode.Away: HVAC_MODE_HEAT,
+    eq3.Mode.Boost: HVAC_MODE_AUTO,
 }
 
 HA_TO_EQ_HVAC = {
-    HVAC_MODE_HEAT: eq3.Mode.Manual,
     HVAC_MODE_OFF: eq3.Mode.Closed,
+    HVAC_MODE_HEAT: eq3.Mode.Manual,
     HVAC_MODE_AUTO: eq3.Mode.Auto,
 }
+HA_TO_EQ_HVAC_SENSOR_CONTROL = {
+    HVAC_MODE_OFF: eq3.Mode.Closed,
+    HVAC_MODE_HEAT: eq3.Mode.Open,
+}
 
 EQ_TO_HA_PRESET = {eq3.Mode.Boost: PRESET_BOOST, eq3.Mode.Away: PRESET_AWAY}
 
 HA_TO_EQ_PRESET = {PRESET_BOOST: eq3.Mode.Boost, PRESET_AWAY: eq3.Mode.Away}
 
+EQ_TO_STRING = {
+    eq3.Mode.Unknown: "unknown",
+    eq3.Mode.Closed: "closed",
+    eq3.Mode.Open: "open",
+    eq3.Mode.Auto: "auto",
+    eq3.Mode.Manual: "manual",
+    eq3.Mode.Away: "away",
+    eq3.Mode.Boost: "boost",
+}
 
-DEVICE_SCHEMA = vol.Schema({vol.Required(CONF_MAC): cv.string})
+EQ_TO_HA_HVAC_ACTION = {
+    eq3.Mode.Open: CURRENT_HVAC_HEAT,
+    eq3.Mode.Closed: CURRENT_HVAC_IDLE,
+}
+
+DEVICE_SCHEMA = vol.Schema(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler